org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/WtpConfigurator.java

```diff
@@ -61,9 +62,14 @@
     public void init(InitializationContext context, IProgressMonitor monitor) {
         // TODO (donat) add required model declarations to the project configurator extension point
         GradleBuild gradleBuild = context.getGradleBuild();
-        this.gradleBuild = (DefaultGradleBuild) gradleBuild;
-        Collection<EclipseProject> eclipseProjects = ModelProviderUtil.fetchAllEclipseProjects(this.gradleBuild, GradleConnector.newCancellationTokenSource(), FetchStrategy.LOAD_IF_NOT_CACHED, monitor);
-        this.locationToProject = eclipseProjects.stream().collect(Collectors.toMap(p -> p.getProjectDirectory(), p -> p));
+        try {
```

Comment:
⭕️ Why this change?

Reply:
It seemed better to use a public API to load models in all configurators. Besides, it hid the problem with model caching: the original code filled the `ModelProvider` cache by accident.